### PR TITLE
Ported hammer based secnario to Nailgun

### PR DESCRIPTION
**Pull request by vijay8451**
Below Scenarios moved to nailgun from hammer:

1. test_client.py
    - Scenario_upgrade_old_client_and_package_installation
    - Scenario_upgrade_new_client_and_package_installation

Note: Testing is pending for these above.

### CODE REVIEW

**Comment by san7ket** [comment]
Test results please

**Comment by vijay8451** [comment]
Test Request:


robottelo]# pytest -v -m pre_upgrade tests/upgrades/test_client.py -k test_pre_scenario_preclient_package_installation
============================================================================ test session starts ============================================================================
platform linux2 -- Python 2.7.5, pytest-3.6.1, py-1.7.0, pluggy-0.6.0 -- /home/vijsingh/my_projects/py2test/bin/python2
cachedir: .pytest_cache
rootdir: /home/vijsingh/my_projects/ROBO/6.4.z/robottelo, inifile:
plugins: services-1.3.0, mock-1.10.0
collected 3 items / 2 deselected                                                                                                                                            

tests/upgrades/test_client.py::Scenario_upgrade_old_client_and_package_installation::test_pre_scenario_preclient_package_installation 

PASSED                          [100%]

================================================================= 1 passed, 2 deselected in 1101.98 seconds =================================================================




robottelo]# pytest -v -s -m post_upgrade tests/upgrades/test_client.py -k test_post_scenario_preclient_package_installation
============================================================================ test session starts ============================================================================
platform linux2 -- Python 2.7.5, pytest-3.6.1, py-1.7.0, pluggy-0.6.0 -- /home/vijsingh/my_projects/py2test/bin/python2
cachedir: .pytest_cache
rootdir: /home/vijsingh/my_projects/ROBO/6.4.z/robottelo, inifile:
plugins: services-1.3.0, mock-1.10.0
collected 3 items / 2 deselected                                                                                                                                            

tests/upgrades/test_client.py::Scenario_upgrade_old_client_and_package_installation::test_post_scenario_preclient_package_installation 


PASSED	  [100%]

================================================================== 1 passed, 2 deselected in 44.54 seconds ==================================================================


robottelo]# pytest -s -v -m post_upgrade tests/upgrades/test_client.py -k test_post_scenario_postclient_package_installation
============================================================================ test session starts ============================================================================
platform linux2 -- Python 2.7.5, pytest-3.6.1, py-1.7.0, pluggy-0.6.0 -- /home/vijsingh/my_projects/py2test/bin/python2
cachedir: .pytest_cache
rootdir: /home/vijsingh/my_projects/ROBO/6.4.z/robottelo, inifile:
plugins: services-1.3.0, mock-1.10.0
collecting 0 items                                                                                                                                                          2018-12-21 15:52:31 - robottelo - INFO - Started setUpClass: robottelo.test/TestCase
collected 3 items / 2 deselected                                                                                                                                            

tests/upgrades/test_client.py::Scenario_upgrade_new_client_and_package_installation::test_post_scenario_postclient_package_installation




PASSED	[100%]


================================================================= 1 passed, 2 deselected in 1054.83 seconds =================================================================
(py2test) [root@vijsingh robottelo]#

File: tests/upgrades/test_client.py [hunk 1]
```diff
@@ -294,14 +292,13 @@ def test_pre_scenario_preclient_package_installation(self):
...
-        client_name = rhel7_client.keys()[0]
         product, yum_repo = create_yum_test_repo(
             product_name=self.prod_name, repo_url=FAKE_REPO_ZOO3, org=self.org)
         update_product_subscription_in_ak(
             product=product, yum_repo=yum_repo, ak=ak, org=self.org)
         time.sleep(30)
-        attach_subscription_to_host_from_satellite(
-            self.org.id, product.name, str(client_name).lower())
+        attach_custom_product_subscription(
+            prod_name=product.name, host_name=rhel7_client.keys()[0])
```
Comment (jyejare) [inline tests/upgrades/test_client.py]: You forgot to ```lower()``` it

Comment (vijay8451) [reply tests/upgrades/test_client.py]: yeah , it's done now :)

**Comment by jyejare** [comment]
Now, you need to cherry-pick this in all branches.

**Comment by san7ket** [comment]
@vijay8451 I have another feedback, Can you re-check this without the sleeps in between, I Know when using CLI we were facing Timeouts, but with API I think we should be able to handle without sleeps

**Comment by jyejare** [comment]
@san7ket +1

**Comment by vijay8451** [comment]
@san7ket +1 , Agree and it's good to remove all sleeps except below as that is requires for goferd to reload
+++++
        kwargs = {'async': True, 'host': self.docker_vm}
        execute(
            docker_execute_command,
            client_container_id,
            'goferd -f',
            **kwargs
        )
        # Holding on for 30 seconds wihle goferd starts
        time.sleep(30)
+++++

Rest are removed now and testing working fine too.

**Pull request merged**